networkx/algorithms/tree/distance_measures.py

```diff
@@ -0,0 +1,129 @@
+import networkx as nx
+
+__all__ = [
+    "tree_centroid",
+]
+
+
+def _subtree_sizes(G, root):
+    """Return a `dict` of subtree sizes in a tree rooted at a given node.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+       A tree.
+
+    root : node
+       A node in `G`.
+
+    Returns
+    -------
+    s : dict
+       Dictionary of subtree sizes keyed on nodes.
+
+    Examples
+    --------
+    >>> _subtree_sizes(nx.path_graph(4), 0)
+    {0: 4, 1: 3, 2: 2, 3: 1}
+
+    >>> _subtree_sizes(nx.path_graph(4), 2)
+    {2: 4, 1: 2, 0: 1, 3: 1}
+    """
+    sizes = {root: 1}
+    stack = [root]
+    for a, b in nx.dfs_edges(G, root):
+        while stack[-1] != a:
+            x = stack.pop()
+            sizes[stack[-1]] += sizes[x]
+        stack.append(b)
+        sizes[b] = 1
+    for b, a in nx.utils.pairwise(reversed(stack)):
+        sizes[a] += sizes[b]
```

Comment:
This trips me up everytime I review now...
```suggestion
    for a, b in nx.utils.pairwise(reversed(stack)):
        sizes[b] += sizes[a]
```

Reply:
I'll go for more descriptive vars.